{{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/ui/base/BaseRecyclerAdapter.kt

```diff
@@ -14,280 +14,287 @@ import {{ cookiecutter.package_name }}.BR
 /**
  * Base recycler adapter for all recycler adapters.
  */
- @Suppress("TooManyFunctions")
-abstract class BaseRecyclerAdapter<T> : RecyclerView.Adapter<BaseRecyclerAdapter<T>.RecyclerHolder>(), Filterable {
+@Suppress("TooManyFunctions")
+abstract class BaseRecyclerAdapter<T> :
+    RecyclerView.Adapter<BaseRecyclerAdapter<T>.RecyclerHolder>(), Filterable {
     private val arrayListFiltered = ArrayList<T>()
-    private val itemTypeNormal = 1
-    private val itemTypeLoader = 2
     private var filteredText = ""
     protected val arrayList = ArrayList<T>()
     protected val previousArrayList = ArrayList<T>()
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerHolder {
-        val binding: ViewDataBinding = if (viewType == itemTypeNormal) {
-            DataBindingUtil.inflate(LayoutInflater.from(parent.context), getLayoutIdForType(viewType), parent, false)
-        } else {
-            DataBindingUtil.inflate(LayoutInflater.from(parent.context), getLayoutIdForLoading(viewType), parent, false)
+    /**
+     * Inner class to set recycler view holder
+     *
+     * @param viewDataBinding   The [ViewDataBinding] instance
+     */
+    inner class RecyclerHolder(
```

Comment:
Put these inner classes at the very beginning of the `BaseRecyclerAdapter` class to group properly

Reply:
@prem-p-simform They are already at the beginning in my opinion, as there are only variables above them. Do you want to move these inner classes above these variables?

Reply:
@vraj-s-shah Yes please move above variables